packages/scheduler/lib/db/migrations/20250501173800_create_groups_table.ts

```diff
@@ -0,0 +1,17 @@
+import type { Knex } from 'knex';
+import { GROUPS_TABLE } from '../../models/tasks.js';
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.raw(`
+            CREATE TABLE ${GROUPS_TABLE} (
+                key varchar(255) PRIMARY KEY,
```

Comment:
ideally I would like the PK here to be referenced as the `group_key` column in the `tasks` table but adding a FK constraint on `tasks` table would lock it. So as a result there will be no FK enforcement by pg. If someone has an idea about how to do it without any live impact on the `tasks` table in production, let me know.